bigchaindb_driver/offchain.py

```diff
@@ -0,0 +1,325 @@
+
+            cid = 0
+            condition = tx['transaction']['conditions'][cid]
+            input_ = {
+                'fulfillment': condition['condition']['details'],
+                'input': {
+                    'cid': cid,
+                    'txid': tx['id'],
+                },
+                'owners_before': condition['owners_after'],
+            }
+
+        Displaying the input on the prompt would look like::
+
+            >>> input_
+            {'fulfillment': {'bitmask': 32,
+              'public_key': '3Cxh1eKZk3Wp9KGBWFS7iVde465UvqUKnEqTg2M',
+              'signature': None,
+              'type': 'fulfillment',
+              'type_id': 4},
+             'input': {'cid': 0,
+              'txid': '57cff2b9490468bdb6d4767a1b07905fdbe18d638d9c7783f639b'},
+             'owners_before': ['3Cxh1eKZk3Wp9KGBWFS7iVde465UvqUKnEqTg2M']}
+
+        To prepare the transfer:
+
+        >>> prepare_transfer_transaction(
+        ...     inputs=input_,
+        ...     owners_after='EcRawy3Y22eAUSS94vLF8BVJi62wbqbD9iSUSUN',
+        ...     asset=tx['transaction']['asset'],
+        ... )
+
+    """
+    if not isinstance(inputs, (list, tuple)):
+        inputs = (inputs, )
+    if not isinstance(owners_after, (list, tuple)):
+        owners_after = [owners_after]
+
+    # NOTE: Needed for the time being. See
+    # https://github.com/bigchaindb/bigchaindb/issues/797
+    if isinstance(owners_after, tuple):
+        owners_after = list(owners_after)
+
+    fulfillments = [
+        BdbFulfillment(Fulfillment.from_dict(input_['fulfillment']),
+                       input_['owners_before'],
+                       tx_input=TransactionLink(**input_['input']))
+        for input_ in inputs
+    ]
+
+    asset = _normalize_asset(asset, is_transfer=True)
+
+    transaction = Transaction.transfer(
+        fulfillments,
+        owners_after,
+        asset,
+        metadata=metadata,
+    )
+    return transaction.to_dict()
+
+
+def fulfill_transaction(transaction, *, private_keys):
+    """
+    Fulfills the given transaction.
+
+    Args:
+        transaction (dict): The transaction to be fulfilled.
+        private_keys (:obj:`str` | :obj:`list` | :obj:`tuple`): One or
+            more private keys to be used for fulfilling the
+            transaction.
+
+    Returns:
+        dict: The fulfilled transaction payload, ready to be sent to a
+        BigchainDB federation.
+
+    """
+    if not isinstance(private_keys, (list, tuple)):
+        private_keys = [private_keys]
+
+    # NOTE: Needed for the time being. See
+    # https://github.com/bigchaindb/bigchaindb/issues/797
+    if isinstance(private_keys, tuple):
+        private_keys = list(private_keys)
+
+    transaction_obj = Transaction.from_dict(transaction)
+    try:
+        signed_transaction = transaction_obj.sign(private_keys)
+    except KeypairMismatchException as exc:
+        raise MissingSigningKeyError('A signing key is missing!') from exc
```

Comment:
Maybe we should note this exception in the docstrings both here and upstream in the driver's static method?